data/profiles/taskrunner.py

```diff
@@ -0,0 +1,71 @@
+# Copyright 2016, EMC, Inc.
+
+"""
+Download and execute python scripts from the RackHD tasks api in a loop
+"""
+
+import urllib2
+import json
+import sys
+from time import sleep
+
+TASK_REQUEST_PERIOD = 5
+TASKS_URI = 'http://<%=server%>:<%=port%>/api/common/tasks/<%=identifier%>'
+
+json_content_type = {"Content-Type": "application/json"}
+
+def download_task_data():
+    """
+    Download a python script from the tasks API
+    """
+    _task_data = urllib2.urlopen(TASKS_URI).read()
+    return json.loads(_task_data)
+
+
+def download_script(downloadUrl):
+    """
+    Download a python script from the tasks API
+    """
+    url = urllib2.urlparse.urljoin('http://<%=server%>:<%=port%>', downloadUrl)
+    print "Downloading script at {}".format(url)
+    script = urllib2.urlopen(url).read()
+    with open('script.py', 'w') as rackhd_script:
+        rackhd_script.write(script)
+
+while True:
```

Comment:
@yyscamper in the Cisco case, we do wrap this in a try/catch, but from the outer script we downloaded via tftp: https://github.com/benbp/on-tftp/blob/cisco-poap/data/templates/cisco-poap.py#L16-L27

However, I'm realizing in the Arista case, we won't be wrapping this. That said, I think if the taskrunner execution fails at this level, my initial thinking is we *don't* want to recover from that, because staying in an infinite loop regardless of what happens is worse than letting the switch continue with its auto-install process (and likely fail that as well since the script will exit with a non-zero code). Perhaps a better approach is to wrap the urllib requests with retry handling, and add a smaller catch block for the `int(task_data["exit"])` statement. What do you think?